pkgs/development/python-modules/diagrams/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+#, poetry
+, bash
+, black
+, jinja2
+, round
+, graphviz
+, inkscape
+, imagemagick
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "diagrams";
+  version = "0.17.0";
+  format = "pyproject";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "mingrammer";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1h5i9xb4fsmy81sznvqycxl712d9k695vznkpx2wm1ki7hvz86s8";
+  };
+
+  preConfigure = ''
+    sed -i "s@#!/bin/bash@$!/${bash}/bin/bash@" autogen.sh
+    ./autogen.sh
+  '';
+
+  # IDFK about python. but regular pip results in an empty python module
+  #installPhase = ''
+    #${poetry}/bin/poetry install
+  #'';
+
+  # the deps specified here below dont exactly match what is in pyproject.toml
+  # I did not investigate whether/why the others are listed yet:
+  #
+  #[tool.poetry.dev-dependencies]
+  #pylint = "^2.4"
+  #rope = "^0.14.0"
+  #isort = "^4.3"
+
+  checkInputs = [ pytest ];
+  nativeBuildInputs = [ black jinja2 inkscape imagemagick round ];
```

Comment:
The `nativeBuildInputs` are only available during the build. That does not seem sensible for most of these.

- black is an optioniated reformatter for python code, we don't need that in packaging at all
- jinja2 is a templating library, needs to be `propagatedBuildInputs`
- inkscape, imagemagick and round … not sure, likely `buildInputs` or `propagatedBuildInputs` when that doesn't work

Reply:
From what I can tell during testing and by digging into the project structure, the only runtime dependency is `graphviz`. 

I am testing with `nix-shell --pure -I nixpkgs=./ -p python3 python3Packages.diagrams python3Packages.graphviz` (from my local nixpkgs clone).

The build procedure goes through the list of providers, some get special processing like AWS:
1. [autogen.sh preprocessing AWS provider resources](https://github.com/mingrammer/diagrams/blob/master/autogen.sh#L45)
2. [calls resource script](https://github.com/mingrammer/diagrams/blob/master/scripts/resource.py#L157)
3. [calls round program](https://github.com/mingrammer/diagrams/blob/master/config.py#L22)

Then the [generate script](https://github.com/mingrammer/diagrams/blob/master/scripts/generate.py#L5) is [run for all providers](https://github.com/mingrammer/diagrams/blob/master/autogen.sh#L52).

* black could possibly be removed. It runs last after code generation; [see ref](https://github.com/mingrammer/diagrams/blob/master/autogen.sh#L61). My arguments for keeping are that I struggled to patch it out, and also it looks nicer if someone browses the code.
* jinja2 is build time, used to generate code. There are not any uses at runtime that I saw. I grepped 'jinja' and [saw one instance in python code](https://github.com/mingrammer/diagrams/blob/master/scripts/generate.py#L5).